src/Frontend/Core.hs

```diff
@@ -163,7 +164,8 @@ type FormHandler p = FormH '[HTML, PlainText] (Frame (FormPageRep p)) (FormPageR
 semanticDiv :: forall m a. (Monad m, Typeable a) => a -> HtmlT m () -> HtmlT m ()
 semanticDiv t = div_ [makeAttribute "data-aula-type" (cs . show . typeOf $ t)]
 
-type DfForm a = forall m. Monad m => DF.Form (Html ()) m a
+type DfFormM m a = DF.Form (Html ()) m a
```

Comment:
Elsewhere we use `SomeThingM` for constraint types.  I assume you needed the unquantified version somewhere, but perhaps we can give up on the quantified version altogether, and always quantify explicitly?  If that's not good, what about `DfForm` for `DfFormM` and `DfFormA` for `DfForm`?  (`A` for "all".  Hm, I still prefer removing `DfForm`.)